### PR TITLE
Fix MinGW warnings (fixes #4731)

### LINKED ISSUE #4731: MinGW warnings
On MinGW, I'll fix them later:
```
/home/deen/isos/ddnet/ddnet-source/src/engine/client/backend_sdl.cpp:791:41: warning: unknown conversion type character ‘z’ in format [-Wformat=]
  791 |         dbg_msg("gfx", "Created OpenGL %zu.%zu context.", (size_t)g_Config.m_GfxOpenGLMajor, (size_t)g_Config.m_GfxOpenGLMinor);
      |                                         ^
/home/deen/isos/ddnet/ddnet-source/src/engine/client/backend_sdl.cpp:791:45: warning: unknown conversion type character ‘z’ in format [-Wformat=]
  791 |         dbg_msg("gfx", "Created OpenGL %zu.%zu context.", (size_t)g_Config.m_GfxOpenGLMajor, (size_t)g_Config.m_GfxOpenGLMinor);
      |                                             ^
/home/deen/isos/ddnet/ddnet-source/src/engine/client/backend_sdl.cpp:791:24: warning: too many arguments for format [-Wformat-extra-args]
  791 |         dbg_msg("gfx", "Created OpenGL %zu.%zu context.", (size_t)g_Config.m_GfxOpenGLMajor, (size_t)g_Config.m_GfxOpenGLMinor);
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

```
/home/deen/isos/ddnet/ddnet-source/src/engine/server/databases/sqlite.cpp: In member function ‘virtual void CSqliteConnection::Print()’:
/home/deen/isos/ddnet/ddnet-source/src/engine/server/databases/sqlite.cpp:233:55: warning: the address of ‘char* sqlite3_expanded_sql(sqlite3_stmt*)’ will never be NULL [-Waddress]
  233 |         if(m_pStmt != nullptr && sqlite3_expanded_sql != nullptr)
      |                                  ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
```

```
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp: In function ‘time_t filetime_to_unixtime(LPFILETIME)’:
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp:2125:18: warning: comparison of integer expressions of different signedness: ‘time_t’ {aka ‘long long int’} and ‘ULONGLONG’ {aka ‘long long unsigned int’} [-W
sign-compare]
 2125 |         return t == li.QuadPart ? t : (time_t)-1;
      |                ~~^~~~~~~~~~~~~~
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp: In function ‘void set_exception_handler_log_file(const char*)’:
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp:4088:50: warning: cast between incompatible function types from ‘FARPROC’ {aka ‘long long int (*)()’} to ‘BOOL (*)(const char*)’ {aka ‘int (*)(const char*)’}
[-Wcast-function-type]
 4088 |                 auto pExceptionLogFilePathFunc = (BOOL APIENTRY(*)(const char *))(GetProcAddress(gs_ExceptionHandlingModule, "ExcHndlSetLogFileNameA"));
      |                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp: At global scope:
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp:226:13: warning: ‘void logger_file_no_newline(const char*, void*)’ defined but not used [-Wunused-function]
  226 | static void logger_file_no_newline(const char *line, void *user)
      |             ^~~~~~~~~~~~~~~~~~~~~~
In function ‘void str_copy(char*, const char*, int)’,
    inlined from ‘void fs_listdir(const char*, FS_LISTDIR_CALLBACK, int, void*)’ at /home/deen/isos/ddnet/ddnet-source/src/base/system.cpp:2153:11:
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp:2616:16: warning: ‘char* strncpy(char*, const char*, size_t)’ output may be truncated copying between 0 and 511 bytes from a string of length 511 [-Wstringop-
truncation]
 2616 |         strncpy(dst, src, dst_size - 1);
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
In function ‘void str_copy(char*, const char*, int)’,
    inlined from ‘void fs_listdir_fileinfo(const char*, FS_LISTDIR_CALLBACK_FILEINFO, int, void*)’ at /home/deen/isos/ddnet/ddnet-source/src/base/system.cpp:2207:11:
/home/deen/isos/ddnet/ddnet-source/src/base/system.cpp:2616:16: warning: ‘char* strncpy(char*, const char*, size_t)’ output may be truncated copying between 0 and 511 bytes from a string of length 511 [-Wstringop-
truncation]
 2616 |         strncpy(dst, src, dst_size - 1);
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by def-**
@Jupeyy @Robyt3 @Zwelf 

## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by Jupeyy** [comment]
fine to me.
for the printf formats we could also force mingw to use ansi standard
something like `-D__USE_MINGW_ANSI_STDIO=1` (never tested it tho, just saw it) tho it probs won't silence the warning as we wrap them with dbg_msg which adds the warning as attribute

File: src/base/system.cpp [hunk 1]
```diff
@@ -2613,7 +2613,14 @@ void str_append(char *dst, const char *src, int dst_size)
...
 {
+#ifdef __MINGW32__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstringop-truncation"
+#endif
 	strncpy(dst, src, dst_size - 1);
+#ifdef __MINGW32__
+#pragma GCC diagnostic pop
+#endif
 	dst[dst_size - 1] = 0; /* assure null termination */
```
Comment (Robyt3) [inline src/base/system.cpp:2624]: Could also use `strncat` instead, like this:

```
dst[0] = '\0';
strncat(dst, src, dst_size - 1);
```

https://stackoverflow.com/a/1258577/1708371

Comment (def-) [reply src/base/system.cpp]: Thanks, done.

**Review by Zwelf — APPROVED** [review]
looks good to me too

**Pull request merged**